src/wire/packet_manager.cpp

```diff
@@ -294,60 +294,41 @@ bool PacketManager::HardcodedExecuteFilter(std::string query_type) {
 }
 
 // The Simple Query Protocol
+// Fix mis-split bug
 void PacketManager::ExecQueryMessage(InputPacket *pkt, const size_t thread_id) {
-  std::string q_str;
-  PacketGetString(pkt, pkt->len, q_str);
+  std::string query;
+  PacketGetString(pkt, pkt->len, query);
+  if (!query.empty()) {
+    std::vector<StatementResult> result;
+    std::vector<FieldInfo> tuple_descriptor;
+    std::string error_message;
+    int rows_affected;
+
+    // execute the query using tcop
+    auto status = traffic_cop_->ExecuteStatement(
+        query, result, tuple_descriptor, rows_affected, error_message,
+        thread_id);
+
+    // check status
+    if (status == ResultType::FAILURE) {
+      SendErrorResponse(
+          {{NetworkMessageType::HUMAN_READABLE_ERROR, error_message}});
+    }
```

Comment:
I think we probably need to have `else` here and put the logic for sending the correct response into this `else` block, so that after sending the error response it does not send the result.